src/main/java/org/gridsuite/filter/server/FilterService.java

```diff
@@ -0,0 +1,243 @@
+import org.gridsuite.filter.server.dto.AbstractFilter;
+import org.gridsuite.filter.server.dto.FilterAttributes;
+import org.gridsuite.filter.server.dto.AbstractGenericFilter;
+import org.gridsuite.filter.server.dto.NumericalFilter;
+import org.gridsuite.filter.server.dto.ScriptFilter;
+import org.gridsuite.filter.server.entities.AbstractFilterEntity;
+import org.gridsuite.filter.server.entities.AbstractGenericFilterEntity;
+import org.gridsuite.filter.server.entities.LineFilterEntity;
+import org.gridsuite.filter.server.entities.NumericFilterEntity;
+import org.gridsuite.filter.server.entities.ScriptFilterEntity;
+import org.gridsuite.filter.server.repositories.LineFilterRepository;
+import org.gridsuite.filter.server.repositories.ScriptFilterRepository;
+import org.gridsuite.filter.server.utils.FilterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Jacques Borsenberger <jacques.borsenberger at rte-france.com>
+ */
+
+interface Repository<FilterEntity extends AbstractFilterEntity, EntityRepository extends JpaRepository<FilterEntity, String>> {
+    EntityRepository getRepository();
+
+    AbstractFilter toDto(FilterEntity filterEntity);
+
+    FilterEntity fromDto(AbstractFilter dto);
+
+    default Optional<AbstractFilter> getFilter(String name) {
+        Optional<FilterEntity> element = getRepository().findById(name);
+        if (element.isPresent()) {
+            return element.map(this::toDto);
+        }
+        return Optional.empty();
+    }
+
+    default Stream<String> getFiltersNames() {
+        return getRepository().findAll().stream().map(AbstractFilterEntity::getName);
+    }
+
+    default FilterEntity insert(AbstractFilter f) {
+        return getRepository().saveAndFlush(fromDto(f));
+    }
+
+    default void deleteAll() {
+        getRepository().deleteAll();
+    }
+
+    default boolean renameFilter(String name, String newName) {
+        if (getRepository().existsById(name)) {
+            FilterEntity filter = getRepository().getOne(name);
+            getRepository().deleteById(name);
+            filter.setName(newName);
+            getRepository().saveAndFlush(filter);
+            return true;
+        }
+        return false;
+    }
+
+    default boolean deleteById(String name) {
+        if (getRepository().existsById(name)) {
+            getRepository().deleteById(name);
+            return true;
+        }
+        return false;
+    }
+}
+
+@Service
+public class FilterService {
+
+    @Autowired
+    FilterService self;
```

Comment:
Here, self is used to auto inject the service for internal @Transactional method calls support
You need to annotate some methods with @Transactional (if only to handle the lazy collections in LineFilterEntity), and calling these methods In Spring cannot be done from the same service instance.

See ContingencyListService in actions-server, or StudyService in study-server

Reply:
it should be for the rename

Reply:
and it is already on insert

Reply:
You can remove 'self' field in FilterService, because you don't use it ...